pennylane/noise/conditionals.py

```diff
@@ -0,0 +1,438 @@
+            f"OpEq({cops_names if len(cops_names) > 1 else cops_names[0]})",
+        )
+
+    def _check_eq_ops(self, x):
+        if not any(not isclass(op) and getattr(op, "arithmetic_depth", 0) for op in self._cond):
+            return _get_ops(x) == self._cops
+
+        try:
+            return _get_ops(x) == self._cops and (
+                _check_with_lc_op(self._cond[0], x)
+                if len(self._cops) == 1
+                else len(x) == len(self._cond)
+                and all(
+                    _check_with_lc_op(_op, _x)
+                    for (_x, _op) in zip(x, self._cond)
+                    if not isclass(_x) and getattr(_x, "arithmetic_depth", 0)
+                )
+            )
+        except:  # pylint: disable = bare-except # pragma: no cover
+            raise ValueError(
+                "OpEq does not support arithmetic operations "
+                "that cannot be converted to a linear combination"
+            ) from None
+
+
+def _get_ops(val):
+    """Computes the class for a given argument from its string name, instance,
+    or a sequence of them.
+
+    Args:
+        val (Union[str, Operation, Iterable]): object to be used
+            for building the wire set.
+
+    Returns:
+        tuple[class]: tuple of :class:`Operation <pennylane.operation.Operation>`
+        classes corresponding to val.
+    """
+    vals = val if isinstance(val, (list, tuple, set, qml.wires.Wires)) else [val]
+    return tuple(
+        (
+            getattr(qml.ops, val)
+            if isinstance(val, str)
+            else (val if isclass(val) else getattr(val, "__class__"))
+        )
+        for val in vals
+    )
+
+
+def _check_with_lc_op(op1, op2):
+    """Helper method for comparing two arithmetic operators using their LinearCombination"""
+    # pylint: disable = unnecessary-lambda-assignment
+    lc_cop = lambda op: qml.ops.LinearCombination(*qml.simplify(op).terms())
+
+    coeffs, op_terms = lc_cop(op1).terms()
+    sprods = [_get_ops(getattr(op_term, "operands", op_term)) for op_term in op_terms]
+
+    def _lc_op(x):
+        coeffs2, op_terms2 = lc_cop(x).terms()
+        sprods2 = [_get_ops(getattr(op_term, "operands", op_term)) for op_term in op_terms2]
+        present = True
+        for coeff, sprod in zip(coeffs2, sprods2):
+            present = sprod in sprods
+            if not present:
+                break
+            p_index = sprods.index(sprod)
+            if not qml.math.equal(coeff, coeffs[p_index]):
+                present = False
+                break
+            coeffs.pop(p_index)
+            sprods.pop(p_index)
+
+        return present
+
+    return _lc_op(op2)
+
+
+def op_in(ops):
+    """Builds a ``Conditional`` as a boolean function for evaluating
+    if a given operation exist in a specified set of operation.
+
+    Args:
+        ops (str, Operation, Union(list[str, Operation])): string
+            representation or instance of the operation.
```

Comment:
Here I was expecting `op_in([qml.RX, qml.Hadamard])`. Is this approach challenging/not advisable?

Reply:
I think I just forgot to mention it, but doing that is possible as well! I'll add it to the docs. 😁